services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/core/utils.py

```diff
@@ -35,6 +36,13 @@ def docker_client() -> Generator[aiodocker.Docker, None, None]:
     docker = aiodocker.Docker()
     try:
         yield docker
+    except aiodocker.exceptions.DockerError as error:
+        logger.warning(
+            "An unexpected Docker error occurred:\n%s", str(traceback.format_exc())
+        )
+        raise HTTPException(
```

Comment:
Much better! But i still think we should somehow find a better code than 500 for this. As I said several times above, this is already the default behaviour of fastapi